pkg/deploy/controller/deployment/controller.go

```diff
@@ -56,9 +56,35 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 
 		deploymentPod, err := c.podClient.createPod(deployment.Namespace, podTemplate)
 		if err != nil {
-			// If the pod already exists, it's possible that a previous CreatePod succeeded but
-			// the deployment state update failed and now we're re-entering.
-			if !kerrors.IsAlreadyExists(err) {
+			if kerrors.IsAlreadyExists(err) {
+				// If the pod already exists, it's possible that a previous CreatePod succeeded but
+				// the deployment state update failed and now we're re-entering.
+				// Ensure that the pod is the one we created by verifying the annotation on it
+				existingPod, err := c.podClient.getPod(deployment.Namespace, deployutil.DeployerPodNameForDeployment(deployment))
+				if err != nil {
+					c.recorder.Eventf(deployment, "failedCreate", "Error getting existing deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
+					return fmt.Errorf("couldn't fetch existing deployer pod for %s: %v", deployutil.LabelForDeployment(deployment), err)
+				}
+				if deployutil.DeploymentNameFor(existingPod) == deployment.Name {
+					// we'll just set the deploymentPod so that pod name annotation
+					// can be set on the deployment below
+					deploymentPod = existingPod
```

Comment:
I lack some domain knowledge.  Is it possible to the image of the deploymentPod that will be used?  If so, shouldn't we verify that this pod is actually running what we expect it to be running?